compiler/rustc_hir_analysis/src/check/wfcheck.rs

```diff
@@ -308,21 +308,12 @@ fn check_item<'tcx>(tcx: TyCtxt<'tcx>, item: &'tcx hir::Item<'tcx>) -> Result<()
         hir::ItemKind::Const(ty, ..) => {
             check_item_type(tcx, def_id, ty.span, UnsizedHandling::Forbid)
         }
-        hir::ItemKind::Struct(_, hir_generics) => {
-            let res = check_type_defn(tcx, item, false);
-            check_variances_for_type_defn(tcx, item, hir_generics);
-            res
-        }
-        hir::ItemKind::Union(_, hir_generics) => {
-            let res = check_type_defn(tcx, item, true);
-            check_variances_for_type_defn(tcx, item, hir_generics);
-            res
-        }
-        hir::ItemKind::Enum(_, hir_generics) => {
-            let res = check_type_defn(tcx, item, true);
-            check_variances_for_type_defn(tcx, item, hir_generics);
-            res
-        }
+        hir::ItemKind::Struct(_, hir_generics) => check_type_defn(tcx, item, false)
```

Comment:
can we use `?` like:

```
check_type_defn()?;
check_variances_for_type_defn()?;
```

then at the bottom we can `Ok(())`?

Seems to simplify control flow a bit

Reply:
nvm this is irrelevant